packages/nuxt/src/pages/utils.ts

```diff
@@ -361,10 +361,16 @@ export function normalizeRoutes (routes: NuxtPage[], metaImports: Set<string> =
     imports: metaImports,
     routes: genArrayFromRaw(routes.map((page) => {
       const route: Record<Exclude<keyof NuxtPage, 'file'>, string> & { component?: string } = Object.create(null)
+      const skipSerializeValue = (val: unknown | unknown[]) => {
+        if (val === undefined) return true
+        if (Array.isArray(val)) return val.length === 0
+        return typeof val !== 'string' && !!val
+      }
```

Comment:
Looks like this utility can live outside the arrow function here - maybe at top level.

Reply:
Those would be nice changes. 👌 

I think we can optimise quite a bit of this.